Обновил классы по БЭМу

feisk opened the pull request.

feisk committed:
Обновил классы по БЭМу

feisk committed:
Обновил картинку на промо(качество 100%), попробовал зафлоатить первый блок

sealeye commented on index.html:
```diff
@@ -11,7 +11,8 @@
   <div class="promo">
     <div class="container">
-      <h1 class"promo__title">Школа сноуборда и горных лыж <span class="promo__title-span">в Санкт-Петербурге</span></h1>
+      <h1 class="promo__title">Школа сноуборда <br>и горных лыж <br><span class="promo__title-span"> 
```
а зачем тебе тут br ?

sealeye replied:
`promo__title-span` так себе класс 
лучше не использовать название тега в классе 
а если ты этот класс применишь не к спану?

feisk replied:
br чтобы перенести текст в заголовке.
Могу сделать через css float и clear
На хелпдей одна из тьюторов сказала что так будет изи, тк текст заголовка в промо вряд-ли будут менять. Не согласен с ней, думаю могут и поменять в реальной жизни. Я сделал, но ожидал, что ты не оценишь)

feisk replied:
Я думаю можно сделать ширину заголовка такой, чтобы текст влезал как в макете

feisk replied:
Правда Громов говорил, что в разных браузерах/системах по разному отрисовывается шрифт, но думаю если сделать запас, то будет норм

sealeye replied:
>br чтобы перенести текст в заголовке

я это ответ просто уровня капитана очевидность :) 

на самом деле использовать br можно когда строка логически закончена, то есть к фразе уже ничего не добавишь 
а тут да, прекрасно подойдет ограничение ширины, либо задать внутренний отступ

sealeye replied:
разница рендера шрифта может быть ну максиму пикселя 4 на весь огромный блок
кстати, можешь это проверить в сафари 
сафари шрифты рендерит жирнее, чем хром на макоси
попробуй сам посмотреть и сравнить

lizzzzzy commented on index.html:
```diff
@@ -55,9 +56,9 @@ <h3 class="services__info-title">Разовое посещение</h3>
             </div>
```
почему здесь сразу блок и его же элемент?
`<a href="#" class="btn__join btn">Записаться</a>`

sealeye commented on index.html:
```diff
@@ -55,9 +56,9 @@ <h3 class="services__info-title">Разовое посещение</h3>
             </div>
           </article>
-          <article class="services__item">
+          <article class="services__item services__item--width">
```
за что отвечает модификатор `services__item--width` ?

sealeye replied:
`services__item` может быть отдельным блоком?

feisk replied:
Ну он позволяет мне задать ему ширину, большую чем у services__item.
Вообще это же модификатор элемента services__item. Или я ошибаюсь?
Еще можно закинуть эти два services__item в контейнер обертку и ему задать ширину, но пока я не совсем понял какой вариант лучше, какой хуже.

feisk replied:
Я сейчас беру размеры из макета и задаю их, ну и смотрю размеры отступов конечно же

feisk replied:
Сейчас сделал services__item для картинки тоже, поменял порядок в html и зафлоатил их подряд, только странно что если убираю высоту в css (245px) то сетка ломается. Почему так может быть?

sealeye replied:
вариант с оберткой явно лучше
потому что использование модификатора предполагает, что просто этот блок чем-то отличается от других

sealeye replied:
про сетку позже поговорим, там может быть миллион причин

sealeye commented on index.html:
```diff
@@ -66,7 +67,7 @@ <h3 class="services__info-title">Краткий курс</h3>
+          <article class="services__item services__item--width">
               <img src="img/service_img_3-btm.jpg" alt="сноубордист" class="services__photo">
             <div class="services__info">
               <h3 class="services__info-title">Полный курс</h3>
```
`h2 class="services__title">Наши услуги</h2>`
`<h2 class="advantages__title">Почему мы</h2>`
если заголовки имеют одинаковый вид, то им нужен одинаковый класс 
(миксовать никто не запрещал)

sealeye replied:
` <a href="#" class="services__more"></a>`
почему кнопка пустая ?
там должен быть хоть какой-нибудь текст 
его потом можно скрыть, но в коде хотя бы понятно что делает этот элемент

sealeye replied:
iframe с картой точно относится к блоку `community` ?

feisk replied:
Сделал title-h2, это ок?

feisk replied:
Думаю карта скорее блок, только если я оставляю ее в блоке community, там же все равно будет принадлежность миксом, верно?

sealeye replied:
`title-h2` не ок
а если так будет выглядеть не заголовок второго уровня?
тут можно main-title , normal-title, usual-title 
ну в общем, надеюсь ты понял по какому принципу наименовать 
не используй названия тега в классе никогда

sealeye commented:
у тебя пока только главная страница с БЭМом?

sealeye commented:
в целом, уже очень хорошо понимаешь структуру наименования и разбивания на блоки

feisk commented:
Да, пока только главная, сейчас буду делать дальше.

feisk committed:
Прописал по БЭМу index и instructors, попробовал сделать сетку в index.

sealeye commented on index.html:
```diff
@@ -36,55 +37,50 @@ <h1 class"promo__title">Школа сноуборда и горных лыж <sp
-              <a href="#" class="services__more"></a>
-            </div>
+        <img src="img/service_img_1.jpg" alt="сноубордист" width="360" height="245" class="services__item">
+        <img src="img/service_img_2-top.jpg" alt="сноубордист" width="360" height="244" class="services__item">
```
почему картинка и article имеют одинаковый класс `services__item` ?
то есть по-твоему это совершенно идентичные блоки, к которым применимы одни и те же стили?

sealeye replied:
на самом деле картинка относится к article

sealeye commented on index.html:
```diff
@@ -36,55 +37,50 @@ <h1 class"promo__title">Школа сноуборда и горных лыж <sp
     <section class="advantages">
       <div class="container">
-        <h2 class="advantages__title">Почему мы</h2>
+        <h2 class="advantages__title title-h2">Почему мы</h2>
```
ну мы же говорили, что `title-h2` плохое название для класса

sealeye commented on index.html:
```diff
@@ -146,44 +142,44 @@ <h2 class="team__title">Команда</h2>
           <p class="community__desc">ICE ACE - не только школа, а целое комьюнити вдохновленных снегом и горами людей, мы без ума от сноуборда и горных лыж и часто делимся важными событиями в социальных сетях, следите за нами на канале Youtube и в Instagram или приходи к нам, мы все расскажем.</p>
-          <iframe src="https://yandex.ru/map-widget/v1/-/CBeNuSuuCB" width="1100" height="360" frameborder="0" class="community__map">Адрес на карте</iframe>
         </article>
+        <iframe src="https://yandex.ru/map-widget/v1/-/CBeNuSuuCB" width="1160" height="360" frameborder="0" class="community__map map">Адрес на карте</iframe>
```
есть смысл обернуть айфрейм в какой-нибудь секшн с классом `map`
а то как ты собираешься управлять стилями для карты?

sealeye commented on instructors.html:
```diff
@@ -10,155 +10,150 @@
+    <section class="instructors">
+      <div class="container">
+        <h1 class="instructors__title title-h2">Инструкторы школы</h1>
+        <div class="instructors__search">
```
а блок поиска не может быть отдельным блоком? 
он что может быть только на странице с инструкторами?

sealeye commented on instructors.html:
```diff
@@ -10,155 +10,150 @@
+            <input type="submit" class="instructors__search-btn">
+          </form>
+        </div>
+        <div class="instructors__gallery">
```
опять же, такая галерея что только на странице инструкторов?

sealeye commented on instructors.html:
```diff
@@ -10,155 +10,150 @@
-      <article class="comment">
-      <h3><a href="" src="">Марина Ильина</a></h3>
-      <p>Костя — отличный тренер! Провела с ним всего одно занятие, чего, в принципе, мне и было достаточно. Я и раньше занималась
+      <article class="rider__comment">
```
а такой комментарий тоже что ли нельзя никуда вынести из данного блока?

sealeye commented:
сейчас поняла, что вам не рассказывают про препроцессоры 
какое же это упущение 
не представляю как тяжело тебе писать стили с таким разбиением по бэму

feisk committed:
Обновил классы

feisk committed:
Доделал классы по БЭМу

feisk committed:
обновил фон - без эффекта, чтобы наложить эффект градиентом как в ТЗ указано

feisk committed:
делаю раскладку блоков на флексах, застрял на team

feisk committed:
Немного поправил теги, прописал ссылки

feisk committed:
Сделал первую страницу на флексах

sealeye commented:
https://isqua.ru/blog/2016/06/23/content-po-sentru-fon-po-shirinie/ 
тебе нужно тянуть все фоновые картинки
а то сейчас всё это обрезается и на больших экранах выглядит некруто 

![screen shot 2018-02-19 at 22 59 35](https://user-images.githubusercontent.com/13489907/36395142-96744812-15c8-11e8-85ed-88832bb1587e.png)
![screen shot 2018-02-19 at 22 59 29](https://user-images.githubusercontent.com/13489907/36395145-99e943a8-15c8-11e8-8d08-d3f992e1c364.png)

sealeye commented:
всегда задавай line-height рядом с font-size 

это касается в первую очередь` .title-usually`
там если заголовок станет многострочным, то текст будет слипатся 
![screen shot 2018-02-19 at 22 38 18](https://user-images.githubusercontent.com/13489907/36394508-a5c0b0f6-15c5-11e8-86d0-5accf9c1b328.png)

если в макете стоит auto, то можно поставить line-height: 1.2

sealeye commented:
![screenshot at feb 19 22-48-58](https://user-images.githubusercontent.com/13489907/36394826-19f5dcd4-15c7-11e8-8f4c-ac6741b58322.png)

вот это выглядит максимально глупо 
ты одну цельную фразу разбиваешь две составляющие, при этом отдельно друг от друга они не имеют смысла 
это один заголовок без вариантов 

а чтобы организовать переносы, используй ограничение ширины или паддинги по бокам 
и неразрывный пробел - &nbsp; для того, чтобы слова не переносились друг от друга никуда

вообще почитай про мнемоники для хтмл и типографику 
у меня в методичке был раздел про это небольшой

sealeye commented:
`.team__info` у тебя задана явная высота 
задавай минимальную лучше 

если потом в этом блоке будет дохрена текста, то он обрежется 
а с минимальной высотой этот блок имеет шанс стать больше и вместить в себя всё

sealeye commented:
почему ты `team__select-btn` позиционируешь абсолютно?
есть же столько более прекрасных способов построить сетку так, чтобы она потом не развалилась

sealeye commented:
у `.promo` тоже вижу явную высоту

sealeye commented:
по бэму у тебя всё ок на всех страницах
да и сетку ты делаешь достаточно аккуратно

feisk committed:
Первая страница на 3 экрана

sealeye commented:
почему на мобильном в `community-gallery` одна фотка, а не 4 ?

sealeye commented:
резины у тебя собственно нет вообще 
потому что `.container` имеет четкие рамки (и они ещё неправильные)
у тебя с 320 до 768 должна быть ширина контейнера 320 
сейчас 768, поэтому на условной ширине 456px у тебя контейнер на 768 не помещается и появляется горизонтальный скролл, а это выглядит скверно всегда 
с 768 до 1200 ширина контейнера должна быть 768

sealeye commented:
если хочешь резинить сайт, то не указывай явную ширину, указывай максимальную и минимальную

sealeye commented:
вместо того, чтобы задавать ширину для .`socials__list` сделай марджины правые для `socials__item`
это будет лучше, потому что если добавятся ещё соц сети, то сетка не развалится

sealeye commented:
с `main-menu` такая же тема 
вам вообще рассказывали про переполнение?

sealeye commented:
всем ховерам можно поставить какой-нибудь transition 
сразу аккуратнее будет смотреться

sealeye commented:
зачем тебе высота для лого?
а если картинка внутри будет другого размера?

на самом деле я знаю зачем
из-за фантомного отступа снизу
это потому что картинка инлайновая и имеет отступ снизу как любая буква в типографике 
если сделать картинку блочной, то никакиах фантомных отступов не будет

sealeye commented:
и задавать одновременно 
```
width: 100%;
    height: 100%;
```
это очень плохой тон
опять же, если вставить картинку других размеров, а родителю задать и высоту и ширину, то картинку расплющит 
достаточно будет
```
width: 100%;
    height: auto;
```

sealeye commented:
и вижу, что грешишь задаванием высоты 
рассмотрим пример `.btn`
![screen shot 2018-03-01 at 02 20 24](https://user-images.githubusercontent.com/13489907/36818665-1e8d5b64-1cf7-11e8-9074-b1e92020d01d.png)
вот что будет, если в кнопке станет чуть больше текста 
поэтому есть смысл регулировать размеры кнопки паддингами, а не высотой

sealeye commented:
ещё я уезжаю вечером в понедельник, поэтому было бы классно допилить с тобой этот проект пока я не уехала
я разгрузила свое время сейчас, так что постараюсь больше помогать 
сейчас из основных проблем это резина, это очень объемная тема на самом деле 
подумаю как тебе про неё рассказать подробнее, если с тезисами про ширины будет непонятно

feisk committed:
Правки по первой странице и верстка большого макета второй страницы

feisk committed:
Правка фонов на первой странице

sealeye commented:
![screenshot at mar 02 17-48-26](https://user-images.githubusercontent.com/13489907/36904566-04cd4f44-1e42-11e8-904b-f4c41189114d.png)
обрати , что некоторым тегам нужно задавать шрифт принудительно

feisk committed:
1 и 2 страница на все макеты, 3 частично готова.
Добавил карусель в галереи на 1 и 2 странице.
Меню на планшет и мобильную версию не добавлял, сейчас там сверху полупрозрачная полоса при скролле.
3 страницу начал делать на sass - стили пока подключил через временный файл "common.css"

feisk committed:
Добавил стили для блока "Course" на странице с расписанием

feisk committed:
Cделал 3 страницу.
Стили добавил в общий файл Style.

feisk committed:
Добавил полифил для picture и заменил filter псевдоэлементами.
